### PR TITLE
Colm/ops 5175

### PR DESCRIPTION
**Pull request by cavster**

### CODE REVIEW

File: app/action/GetJourneyActionFactory.scala [hunk 1]
```diff
@@ -81,4 +82,28 @@ final class GetJourneyActionFactory @Inject() (
...
 
+  def maybeSurveyJourneyActionRefiner(id: SurveyJourneyId): ActionRefiner[Request, SurveyRequest] =
+    new ActionRefiner[Request, SurveyRequest] {
+
+      override protected def refine[A](request: Request[A]): Future[Either[Result, SurveyRequest[A]]] = {
+        implicit val r: Request[A] = request
+        for {
+          maybeJourney <- journeyService.findLatestByJourneyId(id)
+          maybeSurveyRequest <- maybeJourney match {
+            case Some(j) => Future.successful {
```
Comment (rvijayan) [inline app/action/GetJourneyActionFactory.scala:93]: Can this code block go in the yield?

```
     for {
          maybeJourney <- journeyService.findLatestByJourneyId(id)
      } yield  maybeJourney match {
          case Some(j) => Right(SurveyRequest(j.content, j.audit, Option(j.origin), r))
         case None  => Right(SurveyRequest.default)
    }

File: app/paysurvey/journey/ssj/SsjRequest.scala [hunk 2]
```diff
@@ -33,9 +34,42 @@ final case class SsjRequest
...
 }
 
 object SsjRequest {
   implicit val format = Json.format[SsjRequest]
 }
+
+final case class SsjJourneyRequest
+  (
+    origin:         String,
+    returnMsg:      String,
```
Comment (rvijayan) [inline app/paysurvey/journey/ssj/SsjRequest.scala:46]: Do we need to accept a return message? Can we have generic template for survey?

File: app/paysurvey/journey/ssj/SsjResponse.scala [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 import payapi.cardpaymentjourney.model.journey.Url
-import paysurvey.journey.JourneyId
+import paysurvey.journey.SurveyJourneyId
 import play.api.libs.json.{Format, Json}
 
 final case class SsjResponse
   (
-    journeyId: JourneyId,
+    journeyId: SurveyJourneyId,
     nextUrl:   Url
```
Comment (rvijayan) [inline app/paysurvey/journey/ssj/SsjResponse.scala:10]: Why do we need nextUrl here? I thoughts the clients will be building the url for survey with the SurveyJourneyId

**Review by rvijayan — APPROVED** [review]

File: app/paysurvey/journey/ssj/SsjResponse.scala (hunk 3 above)
Comment (cavster) [reply app/paysurvey/journey/ssj/SsjResponse.scala:10]: Its for the consumer to redirect too.This way if we make a change in the URL here we don't have to change it in the calling mircoservices

File: app/action/GetJourneyActionFactory.scala (hunk 1 above)
Comment (cavster) [reply app/action/GetJourneyActionFactory.scala:93]: I am using maybeSurveyRequest below it

File: app/paysurvey/journey/ssj/SsjRequest.scala (hunk 2 above)
Comment (cavster) [reply app/paysurvey/journey/ssj/SsjRequest.scala:46]: This way the caller can specific what the message is

**Pull request merged**